### DISCUSSION TITLE
Is it correct that brunch only exports assets if they are watched?

### DISCUSSION THREAD

**Discussion by macta** [brunch/brunch#1871] [Q&A]
### Description

In my brunch_config.js, I added a dev folder to my exports.conventions like this:

exports.conventions = {
    ignored: ignoredPaths,
    assets: ['site/**/*', 'dev/**/*']
};

And I expected to see some files I put in the dev/* directory to be put in my public folder - however they don't seem to appear?

### Expected behavior

I shouldn't have to do anything else?

### Actual behavior

The files in /dev/* only appear when I also add that directory to my watched paths e.g.

exports.paths = {
    public: pipeline.BUILD,
    watched: [
        pipeline.SRC, pipeline.TEMPLATES, pipeline.SITE, pipeline.PLUGINS, 'scripts', 'dev'
    ]
};

I'm not sure I should need to do that? If I'm not interested in whether they change, but just want them copied over when something else changes, I didn't think I need to adjust my watched list.

### Environment

1. Brunch: *(make sure you're on the latest release!)* 2.10.9
1. Node.js: v7.5.0
1. NPM: *(npm v2 is not supported)* 4.1.2
1. Operating system: OSX
1. Code editor: IntelliJ

**Comment by shvaikalesh** [comment]
Hey @macta, that is a good question!

`paths.watched` are for `chokidar`: they are necessary to find and watch files. So yes, even if changes are not expected, you should specifiy `dev/` there.

Why? `conventions.assets` accepts functions/regexps, so sadly we can't 100% accurately infer watched directories from it. And `chokidar` needs strings only, because performance reasons.

It is current situation, perhaps it will change in future.

**Comment by macta** [comment]
Hi  - thanks for explaining this, but I think my question is actually different (or I could be misunderstanding). The assets I am putting in `dev/` are min.js files and so I'm not expecting them to change and I don't expect `chokidar` to watch them - thats why I left them out of the `paths.watched` setting. I don't understand why `conventions.assets` only works if the same directory is in `paths.watched` - that seems strange to me as the concepts seem different. One is for triggering operations if files change, the other is what to copy. So why can't it just copy if something else changes - or indeed in production mode as part of a CI build, just copy as `chokidar` is not used in that mode?
